openproblems/tasks/chromatin_potential/methods/linear_regression.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+import scanpy as sc
+import os
+from ....tools.decorators import method
+
+
+def _chrom_limit(x):
+    y = x.values
+    if y[-1] == "+":
+        return [y[-3] - 1e5, y[-3] + 1e5]
+    else:
+        return [y[-2] - 1e5, y[-2] + 1e5]
+
+
+def _get_annotation(adata):
+    from pyensembl import EnsemblRelease
+
+    os.system("pyensembl install --release 100 --species mus_musculus")
+    data = EnsemblRelease(100, species="mus_musculus")
+    # get ensemble gene coordinate
+    genes = []
+    for i in adata.var.index.map(lambda x: x.split(".")[0]):
+        try:
+            gene = data.gene_by_id(i)
+            genes.append(
+                [
+                    gene.gene_id,
+                    gene.gene_name,
+                    "chr%s" % gene.contig,
+                    gene.start,
+                    gene.end,
+                    gene.strand,
+                ]
+            )
+        except:
+            genes.append([np.nan, np.nan, np.nan, np.nan, np.nan, np.nan])
+    adata._var = pd.concat(
+        [adata.var, pd.DataFrame(genes, index=adata.var_names)], axis=1
+    )
+
+
+def _atac_genes_score(adata, top_genes=500, threshold=1):
+    import pybedtools
+
+    # basic quality control
+    sc.pp.filter_cells(adata, min_genes=200)
+    sc.pp.filter_genes(adata, min_cells=5)
+
+    adata.var["mt"] = adata.var.gene_short_name.str.startswith(
+        "mt-"
+    )  # annotate the group of mitochondrial genes as 'mt'
+    sc.pp.calculate_qc_metrics(
+        adata, qc_vars=["mt"], percent_top=None, log1p=False, inplace=True
+    )
+
+    adata._inplace_subset_obs(adata.obs.n_genes_by_counts < 2000)
+    adata._inplace_subset_obs(adata.obs.pct_counts_mt < 10)
+    adata.obsm["X_tsne"] = adata.obs.loc[:, ["tsne_1", "tsne_2"]].values
```

Comment:
This definitely is not guaranteed to exist. Why is it here?

Reply:
This key exist for the scicar dataset, but it is not necessary, I'll remove it.